pkg/session/session.go

```diff
@@ -40,18 +41,59 @@ import (
 	"github.com/vmware/govmomi/vim25/methods"
 	"github.com/vmware/govmomi/vim25/soap"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
 
 	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/constants"
 )
 
+const (
+	metricNameSpace            = "session"
+	metricLabelServer          = "server"
+	metricLabelDC              = "dc"
+	metricLabelUsername        = "username"
+	metricLabelOperationType   = "operation"
+	metricLabelGetOperation    = "get"
+	metricLabelCreateOperation = "create"
+	metricLabelDeleteOperation = "delete"
+	metricLabelSessionKey      = "sessionKey"
+)
+
 var (
 	// global Session map against sessionKeys in map[sessionKey]Session.
 	sessionCache sync.Map
 
 	// mutex to control access to the GetOrCreate function to avoid duplicate
 	// session creations on startup.
 	sessionMU sync.Mutex
+
+	// sessionCacheMetric represents a Prometheus GaugeVec (vector of gauges) to track
+	// the number of cached sessions. This metric provides information about the current
+	// count of cached sessions.
+	sessionCacheMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metricNameSpace,
+			Name:      "cached_num",
```

Comment:
```suggestion
			Name:      "session_cached_total",
```

also see naming conventions: https://prometheus.io/docs/practices/naming/

Reply:
*If* we use `capv_client_session` as sub-system this should then be `cached_total`